src/UnitfulRecipes.jl

```diff
@@ -75,7 +75,30 @@ end
 @recipe function f(x::T1, y::T2, f::Function) where {T1<:AVec{<:Union{Missing,<:Quantity}}, T2<:AVec{<:Union{Missing,<:Quantity}}}
     x, y, f.(x',y)
 end
+@recipe function f(f::Function, u::Units)
+    uf = UnitFunction(f, [u])
+    recipedata = RecipesBase.apply_recipe(plotattributes, uf)
+    (_, xmin, xmax) = recipedata[1].args
+    return f, xmin*u, xmax*u
+end
```

Comment:
I like this trick of letting `RecipesBase` handle the limits. Of course it relies a bit on internals of RecipesBase, but I think its still the best approach.